### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by Auryte**

### CODE REVIEW

**Comment by Auryte** [comment]
Screenshot 2022-10-24 181030
Screenshot 2022-10-24 181049

File: client/src/components/ErrorPage/Error.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  test('button is rendering', async () => {
+    render(<Button type="button" children="Return Home" className={ButtonStyles.ErrorBtn} />);
+    expect(screen.getByTestId(testIds.button)).toBeInTheDocument();
+  });
+
+  test('SecondaryPagesLayout is rendering', async () => {
+    render(<SecondaryPagesLayout children={undefined} />);
+    expect(screen.getByTestId(testIds.layout)).toBeInTheDocument();
+  });
```
Comment (JuliaShavyrova) [inline client/src/components/ErrorPage/Error.spec.tsx]: in file 'Error.spec.tsx' you should add tests for Error component only, tests for Button, SecondaryPagesLayout and Main should be placed in there own files

File: client/src/components/ErrorPage/Error.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    render(
+      <MemoryRouter>
+        <Routes>
+          <Route index element={<Main />} />
+        </Routes>
+      </MemoryRouter>
+    );
+    const links: HTMLAnchorElement[] = screen.getAllByRole('link');
+    expect(links[0].textContent).toEqual('Home');
+    expect(links[0].href).toContain('/');
```
Comment (JuliaShavyrova) [inline client/src/components/ErrorPage/Error.spec.tsx]: please add test case for Error component where it should be checked `Return Home` button functionality

Comment (Auryte) [reply client/src/components/ErrorPage/Error.spec.tsx]: but button has no functionality, it just renders. The main functionality covers Link element.

Comment (JuliaShavyrova) [reply client/src/components/ErrorPage/Error.spec.tsx]: no matter, you're testing Error component, that can include anything inside. currently it includes some text and button (and i mean 'button' as UI element, no matter what components are used) with some functionality, so you should write a test to check this functionality. this test should emulate user click the button and check if it was a redirect to home page

File: client/src/components/ErrorPage/Error.spec.tsx [hunk 3]
```diff
@@ -1,46 +1,45 @@
...
+    expect(link.getAttribute('href')).toBe('/');
   });
 
-  test('SecondaryPagesLayout is rendering', async () => {
-    render(<SecondaryPagesLayout children={undefined} />);
-    expect(screen.getByTestId(testIds.layout)).toBeInTheDocument();
-  });
-
-  test('User can navigate to a home page', async () => {
+  test('when button clicks, there is link to "/"', async () => {
```
Comment (Auryte) [inline client/src/components/ErrorPage/Error.spec.tsx]: I did the tests according to info I found " jsdom is simulating a browser but it has some limitations. One of these limitations is the fact that you can't change the location. If you want to test that your link works I suggest to check the href attribute of your <a>.  Is it right?

Comment (JuliaShavyrova) [reply client/src/components/ErrorPage/Error.spec.tsx]: you have client-side routing in your app, that's managed with react-router library (remember, you're wrapping the whole app with BrowserRouter/HashRouter or else component imported from react-router library). so you need to do the same in the test, like here:
https://stackoverflow.com/questions/70974339/testing-navigation-with-react-testing-library

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**